ui/src/treeDataProvider/AzureParentNode.ts

```diff
@@ -69,19 +69,21 @@ export class AzureParentNode<T extends IAzureParentTreeItem = IAzureParentTreeIt
             .sort((n1: AzureNode, n2: AzureNode) => n1.treeItem.label.localeCompare(n2.treeItem.label));
     }
 
-    public async pickChildNode(expectedContextValue: string, ui: IUserInterface): Promise<AzureNode> {
+    public async pickChildNode(expectedContextValues: string[], ui: IUserInterface): Promise<AzureNode> {
         if (this.treeItem.pickTreeItem) {
             const children: AzureNode[] = await this.getCachedChildren();
-            const treeItem: IAzureTreeItem | undefined = this.treeItem.pickTreeItem(expectedContextValue);
-            if (treeItem) {
-                const node: AzureNode | undefined = children.find((n: AzureNode) => n.treeItem.id === treeItem.id);
-                if (node) {
-                    return node;
+            for (const val of expectedContextValues) {
+                const treeItem: IAzureTreeItem | undefined = this.treeItem.pickTreeItem(val);
+                if (treeItem) {
+                    const node: AzureNode | undefined = children.find((n: AzureNode) => n.treeItem.id === treeItem.id);
+                    if (node) {
+                        return node;
+                    }
                 }
             }
         }
 
-        const pick: PickWithData<GetNodeFunction> = await ui.showQuickPick<GetNodeFunction>(this.getQuickPicks(), localize('azFunc.selectNode', 'Select a {0}', this.treeItem.childTypeLabel));
+        const pick: PickWithData<GetNodeFunction> = await ui.showQuickPick<GetNodeFunction>(this.getQuickPicks(expectedContextValues), localize('azFunc.selectNode', 'Select a {0}', this.treeItem.childTypeLabel));
```

Comment:
What will display to the user in the placeholder?  The first treeItem's childTypeLabel?

Reply:
Not sure what you mean by 'the first'. There's only one childTypeLabel at this point in the code (`this.treeItem.childTypeLabel`). For example, if `this` is a Cosmos DB Account, the childTypeLabel is 'Database' and 'Select a Database' will be shown as the placeholder

For your other question, the Cosmos DB extension has several commands like 'Open In Portal' that apply to Cosmos DB Accounts that have different contextValues (mongoDatabaseAccount, graphDatabaseAccount, docDBDatabaseAccount, etc.) even though they're at the same level in the tree

Reply:
Oh, I misunderstood what treeType.childTypeLabel was.  That makes sense.

And also 👍 to your other answer.